model/config.go

```diff
@@ -434,6 +435,10 @@ func (s *ServiceSettings) SetDefaults() {
 		s.Forward80To443 = NewBool(false)
 	}
 
+	if s.TrustedProxyIPHeader == nil {
```

Comment:
We're close to merging https://github.com/mattermost/mattermost-server/pull/10551 -- could we get the version in `master` to adopt the semantics of that PR for deciding which way to flip the defaults? (I realize we'll need to re-create your current code for the cherry-pick).

Reply:
This will need to go into 5.12, so ideally it should be merged today/tomorrow. Is there any timeline for #10551?

Reply:
The problem here is that without the defaults.json this will be the default for new installs. Is this a problem solved by https://github.com/mattermost/mattermost-server/pull/10551?

Reply:
Go ahead and merge -- I'll work with @reflog to iterate on the changes in the linked PR instead.

Reply:
/Just/ missed your comment, @crspeller -- the changes in https://github.com/mattermost/mattermost-server/pull/10551 should be fungible with respect to our current use of `default.json`. The code would look something like:

```go
if isUpdate {
    s.TrustedProxyIPHeader = []string{HEADER_FORWARDED, HEADER_REAL_IP}
} else {
    s.TrustedProxyIPHeader = ""
}
```

where `isUpdate` is `true` whenever `ServiceSettings.SiteURL` is non-nil.